maven-scm-api/src/main/java/org/apache/maven/scm/command/changelog/AbstractChangeLogCommand.java

```diff
@@ -96,11 +122,25 @@ public ScmResult executeCommand( ScmProviderRepository repository, ScmFileSet fi
 
         if ( versionOnly )
         {
-            return executeChangeLogCommand( repository, fileSet, version, datePattern );
+            if ( limit == null )
+            {
+                return executeChangeLogCommand( repository, fileSet, version, datePattern );
+            }
+            else
+            {
+                return executeChangeLogCommand( repository, fileSet, version, datePattern, limit );
```

Comment:
Why this if else pain? Why not pass a null and the short method call the longer one? A concrete implementation can still throw an exception with a non-null arg has been passed.

Reply:
OK. Will look into this and change it as requested.